onnxruntime/core/providers/cuda/tensor/quantize_linear.cc

```diff
@@ -42,41 +24,26 @@ Status QuantizeLinear<T>::ComputeInternal(OpKernelContext* ctx) const {
 
   const auto& x_shape = x->Shape();
 
-  const float* input = x->template Data<float>();
+  const CudaU* input = reinterpret_cast<const CudaU*>(x->template Data<U>());
   T* output = y->template MutableData<T>();
 
+  // TO DO: support per-channel
   ORT_ENFORCE(IsScalarOr1ElementVector(y_scale), "x_scale must be a scalar or 1D tensor of size 1.");
```

Comment:
typo message x->y in these two lines

Reply:
Not related to this PR. From [onnx spec](https://github.com/onnx/onnx/blob/master/docs/Operators.md#QuantizeLinear), the y_zero_point is optional, but current implementation requires it.